decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
Ok, now I understand. For the sake of going for the path of least surprise: Maybe we could write ad-hoc code for the `ParticipatoryProcessController` instead, maybe shared with a common module with `NeedsParticipatoryProcess` (or something of the likes)?